dask/array/slicing.py

```diff
@@ -571,6 +585,42 @@ def insert_many(seq, where, val):
     return tuple(result)
 
 
+@memoize
+def _expander(where):
+    if not where:
+        def expand(seq, val):
+            return seq
+        return expand
+    else:
+        decl = """def expand(seq, val):
+            return ({left}) + tuple({right})
+        """
+        left = []
+        j = 0
+        for i in range(max(where) + 1):
+            if i in where:
+                left.append("val, ")
+            else:
+                left.append("seq[%d], " % j)
+                j += 1
+        right = "seq[%d:]" % j
+        left = "".join(left)
+        decl = decl.format(**locals())
+        ns = {}
+        exec(compile(decl, "<dynamic>", "exec"), ns, ns)
+        return ns['expand']
+
+
+def expander(where):
```

Comment:
I don't think so, unless someone has a need for a large number of locations for `newaxis` in their indices.